File: src/domain/candle/candle.py
from dataclasses import dataclass
from sqlalchemy import Column, Text, Float,  BigInteger
from .. import Base

@dataclass()
class Candle(Base):
    __tablename__ = "candles"

    symbol: str = Column(Text, primary_key=True)
    timestamp: int = Column(BigInteger, primary_key=True)
    open: float = Column(Float)
    high: float = Column(Float)
    low: float = Column(Float)
    close: float = Column(Float)
    volume: float = Column(Float)

    def __str__(self) -> str:
        return f"<Candle(symbol={self.symbol}" \
            f"timestamp={self.timestamp}, " \
            f"open={self.open}, " \
            f"high={self.high}, " \
            f"low={self.low}, " \
            f"close={self.close}, " \
            f"volume={self.volume})> "

File: src/Domain/src/Candle/Candle.cs
namespace Domain.Candle;
public class Candle
{
    public Candle(Symbol symbol, long timestamp, float open, float high, float low, float close, float volume)
    {
        this.Symbol = symbol;
        this.Timestamp = timestamp;
        this.Open = open;
        this.High = high;
        this.Low = low;
        this.Close = close;
        this.Volume = volume;
    }

    public Symbol Symbol { get; private set; }
    public long Timestamp { get; private set; }
    public float Open { get; private set; }
    public float High { get; private set; }
    public float Low { get; private set; }
    public float Close { get; private set; }
    public float Volume { get; private set; }
}
